### PR TITLE
Issue #5208: Add a 'Save and configure permissions' button to the 'Add role' form

### LINKED ISSUE #5208: [UX] Add a 'Save and configure permissions' button to the 'Add role' form
This is a sibling issue of #999

In the "Add content type" form we have this:

image

In the "Add vocabulary" form we are adding this:

image

In the "Add role" form we have this:

image

How about "Save and configure permissions" and "Save role"?

### PR DESCRIPTION
**Pull request by klonos**
Fixes https://github.com/backdrop/backdrop-issues/issues/5208

### CODE REVIEW

File: core/modules/user/user.admin.inc [hunk 1]
```diff
@@ -661,13 +661,23 @@ function user_admin_role($form, &$form_state, $role = NULL) {
...
+      '#value' => t('Save and set permissions'),
+      '#weight' => 1,
+    );
+  }
   $form['actions']['cancel'] = array(
     '#type' => 'link',
     '#href' => 'admin/config/people/roles',
     '#title' => t('Cancel'),
+    '#options' => array('attributes' => array('class' => array('form-cancel'))),
+    '#weight' => 15,
```
Comment (jenlampton) [inline core/modules/user/user.admin.inc]: why the weight?

File: core/modules/user/user.admin.inc [hunk 2]
```diff
@@ -661,13 +661,23 @@ function user_admin_role($form, &$form_state, $role = NULL) {
...
     '#type' => 'submit',
     '#value' => t('Save role'),
+    '#weight' => 2,
   );
+  // Only show the 'Save and set permissions' button when creating a new role.
+  if (empty($role)) {
+    $form['actions']['continue'] = array(
+      '#type' => 'submit',
+      '#value' => t('Save and set permissions'),
+      '#weight' => 1,
```
Comment (jenlampton) [inline core/modules/user/user.admin.inc]: If we move this button above the `Save role` button we won't need the weights.

File: core/modules/user/user.admin.inc (hunk 1 above)
Comment (klonos) [reply core/modules/user/user.admin.inc]: I must have been playing with something on my local 🤷🏼  ...removed with [`11b1eee` (#3728)](https://github.com/backdrop/backdrop/pull/3728/commits/11b1eee42dcac2f37c58368a596ad72f72cee4be)

File: core/modules/user/user.admin.inc (hunk 2 above)
Comment (klonos) [reply core/modules/user/user.admin.inc]: Yes 👍🏼 ...fixed with [`11b1eee` (#3728)](https://github.com/backdrop/backdrop/pull/3728/commits/11b1eee42dcac2f37c58368a596ad72f72cee4be)

File: core/modules/user/user.admin.inc [hunk 3]
```diff
@@ -684,6 +691,7 @@ function user_admin_role_submit($form, &$form_state) {
     );
     user_role_save($role);
     backdrop_set_message(t('The %role role has been added.', array('%role' => $role->label)));
+    $redirect = 'admin/config/people/permissions';
```
Comment (docwilmot) [inline core/modules/user/user.admin.inc]: There is no logic to prevent the form going here if the plain 'save role' btn is clicked.

Comment (klonos) [reply core/modules/user/user.admin.inc]: Thank you @docwilmot!! ...how could I miss that 🤦🏼 

Fixed with: https://github.com/klonos/backdrop/commit/a3d67674445a3c45fd4219a2c0d1ab431ae7bf87

**Comment by klonos** [comment]
Random test failure this time: https://github.com/backdrop/backdrop-issues/issues/5230

**Comment by bugfolder** [comment]
Code reviewed and tested on Tugboat. LGTM.

File: core/modules/user/user.admin.inc [hunk 4]
```diff
@@ -676,6 +683,9 @@ function user_admin_role($form, &$form_state, $role = NULL) {
  * Form submit handler for the user_admin_role() form.
  */
 function user_admin_role_submit($form, &$form_state) {
+  //Set the default redirect.
```
Comment (quicksketch) [inline core/modules/user/user.admin.inc]:
```suggestion
  // Set the default redirect.
```

**Comment by quicksketch** [comment]
Merged into 1.x and 1.20.x. Thanks!

**Pull request merged**